server/genrest/compliance.go

```diff
@@ -55,8 +55,8 @@ func (backend *RESTBackend) HandleRepeatDataBody(w http.ResponseWriter, r *http.
 	// Intentional: Field values in the URL path override those set in the body.
 	var jsonReader bytes.Buffer
 	bodyReader := io.TeeReader(r.Body, &jsonReader)
-	rBytes := make([]byte, r.ContentLength)
-	if _, err := bodyReader.Read(rBytes); err != nil && err != io.EOF {
+	rBytes, err := io.ReadAll(bodyReader)
+	if err != nil && err != io.EOF {
```

Comment:
If switching to ReadAll we no longer need to check for io.EOF. just `err != nil`. From the docs: https://pkg.go.dev/io#ReadAll

" A successful call returns err == nil, not err == EOF. Because ReadAll is defined to read from src until EOF, it does not treat an EOF from Read as an error to be reported."

Reply:
Fixed.